gui/executionGraphGui/client/apps/eg/src/app/components/file-browser/file-browser.component.ts

```diff
@@ -110,16 +146,38 @@ export class FileBrowserComponent implements OnInit {
     });
   }
 
+  private saveFile(path: string, checkOverwrite: boolean) {
+    assertThat(this.mode === FileBrowserMode.Save);
+
+    const showOverwrite = checkOverwrite ? this.checkOverwrite(path) : false;
+
+    let save = false;
+
+    const dialogRef = this.dialog.open(ConfirmationDialogComponent, {
+      minWidth: '10%',
+      data: {
+        title: showOverwrite ? 'Overwrite?' : 'Save?',
+        question: showOverwrite ? `Do you really want to overwrite the path '${path}' ?` : `Saving to path: '${path}' ?`
+      }
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        save = true;
+      }
+    });
+
+    if (save) {
+      this.logger.debug(`Save file '${path}'`);
+      this.fileActionSave.emit(path);
+    }
+  }
+
   public isFileOpenable(file: FileInfo) {
     return file.name.endsWith('.eg');
   }
 
-  private deleteFile(path: FileInfo | DirectoryInfo) {
-    if (this.isFileOpenable(path)) {
-      this.logger.debug(`Deleting path '${path.path}'`);
-    } else {
-      this.logger.error('Programming Error!');
-      throw new Error('Error!');
-    }
+  public isFileNameCorrect(fileName: string) {
+    // https://regex101.com/r/Owqjpb/1
+    return fileName.match(/^(?<!\.)\w([^\\/]+)\.eg$/gm).length > 0;
```

Comment:
Create a Validator for this... (see reactive forms)

Reply:
Makes sense, no idea how to do it, -> later :-)

Reply:
happy for a suggestion =)

Reply:
-> merge back. This can be done in a second step...